### PR TITLE
Fetchure/student controller test (Student get by id method)

### PR DESCRIPTION
**Pull request by NevarDima**
added Student get by id test

### CODE REVIEW

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+class StudentControllerTest extends DatabaseSpecification {
+
+    private TestRestTemplate restTemplate = new TestRestTemplate();
+    private HttpHeaders headers = new HttpHeaders();
+
+    static String setupContent
+    static String cleanupContent
+
+//  выполняется 1 раз перед всеми тестами (подгрузка ресурсов)
+    def setupSpec() {
```
Comment (skalashynski) [inline src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Remove Russian comments

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    @SuppressWarnings("unused")
+    def setup() {
+        //used 1 year expiration token
+        headers.add("Authorization"
+                , "Bearer eyJhbGciOiJIUzI1NiJ9.eyJqdGkiOiJpZF92YWx1ZSIsImlhdCI6MTYzMTcxNzM2MiwiYXV0aG9yaXRpZXMiOlt7ImF1dGhvcml0eSI6IlJPTEVfU1RVREVOVCJ9XSwic3ViIjoiU2VyZ2lLMyIsImlzcyI6IlNwcmluZ19EZW1vX0FQSV9TZXJ2aWNlIiwiZXhwIjoxNjYzMjUzMzYyfQ.D4EVYJu6Cyb82M7vs1QphF11pQYWIgLo57BbHbfgeKQ")
+        sql.execute(setupContent)
+    }
+
+//  выполняется после каждого теста (очистка таблицы)
+    @SuppressWarnings("unused")
```
Comment (skalashynski) [inline src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Remove Russian comments

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    def cleanup() {
+        sql.execute(cleanupContent)
+    }
+
+//  выполняется после всех тестов 1 раз (освобождает ресурсы)
+//    def cleanupSpec() {
+//    }
+
```
Comment (skalashynski) [inline src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Remove Russian comments

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -2,7 +2,7 @@ application:
   jwt:
     secretKey: "secure_secure_secure_secure_secure_secure_secure_secure_secure_secure_secure_secure_secure"
     tokenPrefix: "Bearer "
-    tokenExpirationSeconds: 86400
+    tokenExpirationSeconds: 86400 #31536000(for token for integration tests)
     issuer: "Spring_Demo_API_Service"
```
Comment (skalashynski) [inline src/main/resources/application.yml]: Bad choice to set comments in the property file, remove it

File: build.gradle [hunk 5]
```diff
@@ -59,7 +59,8 @@ dependencies {
             'org.springframework.boot:spring-boot-starter-data-jpa',
             'org.springframework.boot:spring-boot-starter-security',
             'io.jsonwebtoken:jjwt-impl:0.11.2',
-            'org.codehaus.groovy:groovy-all:3.0.5'
+            'org.codehaus.groovy:groovy-all:3.0.5',
+            'com.fasterxml.jackson.core:jackson-databind:2.12.5'
     )
```
Comment (skalashynski) [inline build.gradle]: Do we really need additional libraries, if the 'port problem' was resolved?

File: .jpb/persistence-units.xml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+<project version="4">
+  <component name="PersistenceUnitSettings">
+    <persistence-units>
+      <persistence-unit name="Default">
+        <packages>
+          <package value="com.skalashynski.spring.springboot" />
+        </packages>
+      </persistence-unit>
+    </persistence-units>
+  </component>
```
Comment (skalashynski) [inline .jpb/persistence-units.xml]: Question: let me know, what's the purpose of using persistence-units.xml

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    @Unroll
+    def void "Student get by id"(Long id) {
+        when:
```
Comment (skalashynski) [inline src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: For Spock testing use please #field in the method name
https://www.jetbrains.com/idea/guide/tutorials/writing-tests-with-spock/data-tables/
16:00

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy (hunk 1 above)
Comment (NevarDima) [reply src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: translated

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy (hunk 2 above)
Comment (NevarDima) [reply src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: translated

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy (hunk 3 above)
Comment (NevarDima) [reply src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: translated

File: src/main/resources/application.yml (hunk 4 above)
Comment (NevarDima) [reply src/main/resources/application.yml]: removed

File: build.gradle (hunk 5 above)
Comment (NevarDima) [reply build.gradle]: The problem was resolved and didn't depend on this library.Dependency was deleted.

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy (hunk 7 above)
Comment (NevarDima) [reply src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Corrected according to recommendations

File: .jpb/persistence-units.xml (hunk 6 above)
Comment (NevarDima) [reply .jpb/persistence-units.xml]: We don't have any purpose to use this file. It was created automatically when the idea asked to configure the JPA plugin. persistence-units.xml was deleted

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+@ContextConfiguration
+class StudentControllerTest extends DatabaseSpecification {
+
+    private TestRestTemplate restTemplate = new TestRestTemplate();
+    private HttpHeaders headers = new HttpHeaders();
+
+    static String setupContent
+    static String cleanupContent
+
+//  setup resources (once before all tests)
```
Comment (skalashynski) [inline src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Remove comments

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    def setupSpec() {
+        cleanupContent = ResourceUtils.getResourceContents('cleanup.sql')
+        setupContent = ResourceUtils.getResourceContents('data.sql')
+    }
+
+//  do before each test
+    @SuppressWarnings("unused")
```
Comment (skalashynski) [inline src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Remove comments

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    def setup() {
+        //used 1 year expiration token
+        headers.add("Authorization"
+                , "Bearer eyJhbGciOiJIUzI1NiJ9.eyJqdGkiOiJpZF92YWx1ZSIsImlhdCI6MTYzMTcxNzM2MiwiYXV0aG9yaXRpZXMiOlt7ImF1dGhvcml0eSI6IlJPTEVfU1RVREVOVCJ9XSwic3ViIjoiU2VyZ2lLMyIsImlzcyI6IlNwcmluZ19EZW1vX0FQSV9TZXJ2aWNlIiwiZXhwIjoxNjYzMjUzMzYyfQ.D4EVYJu6Cyb82M7vs1QphF11pQYWIgLo57BbHbfgeKQ")
+        sql.execute(setupContent)
+    }
+
+//  do after each test (clean resources)
```
Comment (skalashynski) [inline src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Remove comments

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+    @SuppressWarnings("unused")
+    def cleanup() {
+        sql.execute(cleanupContent)
+    }
+
+//  clean resources (once after all tests)
+//    def cleanupSpec() {
+//    }
+
```
Comment (skalashynski) [inline src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Remove comments

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+//  do before each test
+    @SuppressWarnings("unused")
+    def setup() {
+        //used 1 year expiration token
+        headers.add("Authorization"
+                , "Bearer eyJhbGciOiJIUzI1NiJ9.eyJqdGkiOiJpZF92YWx1ZSIsImlhdCI6MTYzMTcxNzM2MiwiYXV0aG9yaXRpZXMiOlt7ImF1dGhvcml0eSI6IlJPTEVfU1RVREVOVCJ9XSwic3ViIjoiU2VyZ2lLMyIsImlzcyI6IlNwcmluZ19EZW1vX0FQSV9TZXJ2aWNlIiwiZXhwIjoxNjYzMjUzMzYyfQ.D4EVYJu6Cyb82M7vs1QphF11pQYWIgLo57BbHbfgeKQ")
+        sql.execute(setupContent)
+    }
```
Comment (skalashynski) [inline src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Remove comment and create constant variable for token.

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy (hunk 8 above)
Comment (NevarDima) [reply src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Removed

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy (hunk 9 above)
Comment (NevarDima) [reply src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Removed

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy (hunk 10 above)
Comment (NevarDima) [reply src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Removed

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy (hunk 11 above)
Comment (NevarDima) [reply src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Removed

File: src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy (hunk 12 above)
Comment (NevarDima) [reply src/test/groovy/com/skalashynski/spring/springboot/web/controller/StudentControllerTest.groovy]: Moved value of token variable to application-test.yml

**Review by skalashynski — APPROVED** [review]

**Pull request merged**